clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractFetch.java

```diff
@@ -456,6 +457,36 @@ protected Map<Node, FetchSessionHandler.FetchRequestData> prepareFetchRequests()
         return fetchable.entrySet().stream().collect(Collectors.toMap(Map.Entry::getKey, e -> e.getValue().build()));
     }
 
+    private Set<Integer> nodesWithBufferedPartitions(Set<TopicPartition> buffered, long currentTimeMs) {
+        Set<Integer> nodesWithBufferedPartitions = new HashSet<>();
+
+        for (TopicPartition partition : buffered) {
+            if (!subscriptions.isAssigned(partition)) {
```

Comment:
Can we add test cases for all of the partitions we skip?

Reply:
Will do.

Reply:
Done.